File: modulo_2_SI/2.4/prac1_jeb_drg/src/main/java/es/codeurjc/shoppingCart/domain/Product.java
package es.codeurjc.shoppingCart.domain;

public class Product {

    private Long id;
    private String name;
    private String description;

    public Product() {
    }

    public Product(String name, String description) {
        this.name = name;
        this.description = description;
    }
}

File: modulo_2_SI/2.4/prac1_jeb_drg/Node/src/data/infrastructure/db/schemas/Product.js
const moment = require('moment');
const mongoosePaginate = require('mongoose-paginate');

function create(mongoose) {

    const productSchema = mongoose.Schema({
        name: {
            type: String,
            required: true,
        },
        description: String,
    });


    productSchema.pre('save', (next) => {
        this.created = moment().toJSON();
        return next();
    });

    productSchema.index({ created: -1 });

    productSchema.plugin(mongoosePaginate);

    return mongoose.model('Product', productSchema);
}

module.exports = create;
